data/tekton-pipelines/windows-bios-installer-pipeline.yaml

```diff
@@ -0,0 +1,273 @@
+  tasks:
+    - name: create-vm
+      params:
+        - name: runStrategy
+          value: RerunOnFailure
+        - name: "manifest"
+          value: |-          
+            apiVersion: kubevirt.io/v1
+            kind: VirtualMachine
+            metadata:
+              generateName: "windows-bios-automation-"
+            spec:
+              instancetype:
+                kind: VirtualMachineClusterInstancetype
+                name: n1.large
+              preference:
+                kind: VirtualMachineClusterPreference
+                name: windows.10
+              dataVolumeTemplates:
+                - apiVersion: cdi.kubevirt.io/v1beta1
+                  kind: DataVolume
+                  metadata:
+                    name: installcdrom
+                  spec:
+                    storage:
+                      resources:
+                        requests:
+                          storage: "7Gi"
+                    source:
+                      http:
+                        url: "$(params.winImageDownloadURL)"
+              running: true
+              template:
+                spec:
+                  domain:
+                    devices:
+                      disks:
+                        - cdrom:
+                            bus: sata
+                          name: installcdrom
+                          bootOrder: 2
+                        - disk:
+                            bus: virtio
+                          name: rootdisk
+                          bootOrder: 1
+                        - cdrom:
+                            bus: sata
+                          name: virtiocontainerdisk
+                        - cdrom:
+                            bus: sata
+                          name: sysprep
+                  terminationGracePeriodSeconds: 3600
+                  volumes:
+                  - name: installcdrom
+                    dataVolume:
+                      name: installcdrom
+                  - name: virtiocontainerdisk
+                    containerDisk:
+                      image: $(params.virtioContainerDiskName)
+                  - name: sysprep
+                    sysprep:
+                      configMap: 
+                        name: $(params.autounattendConfigMapName)
+                  - dataVolume:
+                      name:  $(tasks.create-vm-root-disk.results.name)
+                    name: rootdisk
+      taskRef:
+        kind: Task
+        name: create-vm-from-manifest
+      runAfter:
+        - create-vm-root-disk
+    - name: wait-for-vmi-status
+      params:
+        - name: vmiName
+          value: $(tasks.create-vm.results.name)
+        - name: successCondition
+          value: status.phase == Succeeded
+        - name: failureCondition
+          value: status.phase in (Failed, Unknown)
+      runAfter:
+        - create-vm
+      timeout: 2h
+      taskRef:
+        kind: Task
+        name: wait-for-vmi-status
+    - name: create-base-dv
```

Comment:
All gets complicated, because of the disk name and multiple runs use case. In templates, the disk name was derived from VM name and vm name was generated with some random hash. So you could run the pipeline multiple times. But while creating VM from VM manifest, the datavolume can have only a single hardcoded name (it might use generateName, but then we could not copy it to kubevirt-os-images, because the pvc will have different name than VM and vm name and namespace is the only information we get from create-vm task).